File: tests/fixtures/deep-analysis/java/tests/UserApiTest.java
import io.restassured.response.Response;
import org.junit.jupiter.api.Test;
import static io.restassured.RestAssured.given;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class UserApiTest {
    private static final String USERS_PATH = "/users";
    private static final String BASE = "";

    @Test
    public void testGetUsers() {
        Response response = given().when().get(USERS_PATH).then().extract().response();
        assertEquals(200, response.getStatusCode());
    }

    @Test
    public void testGetUserById() {
        String path = BASE + "/users/" + "123";
        Response response = given().when().get(path).then().extract().response();
        assertEquals(200, response.getStatusCode());
    }
}

File: tests/fixtures/deep-analysis/kotlin/tests/UserApiTest.kt
/**
 * Kotlin deep-analysis fixture — UserApiTest.kt
 *
 * Demonstrates various endpoint resolution patterns that the deep analyzer
 * should resolve: enum references, constants, template expressions, and
 * RestAssured assertion linkage.
 */

import io.restassured.RestAssured.given
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

enum class Routes(val path: String) {
    USERS("/users"),
    USER_BY_ID("/users/{id}"),
    ORDERS("/orders")
}

const val BASE_URL = ""

class UserApiTest {

    @Test
    fun `test list users via enum`() {
        val response = given().get(Routes.USERS.path)
        assertEquals(200, response.statusCode)
    }

    @Test
    fun `test list users via constant`() {
        val response = given().get(BASE_URL + Routes.USERS.path)
        response.then().statusCode(200)
    }

    @Test
    fun `test get user by id via enum`() {
        val response = given().get(Routes.USER_BY_ID.path)
        response.then().statusCode(200)
    }

    @Test
    fun `test create user via literal`() {
        val response = given().post("/users")
        assertEquals(201, response.statusCode)
    }

    @Test
    fun `test create order`() {
        val response = given().post(Routes.ORDERS.path)
        assertEquals(201, response.statusCode)
    }
}
